src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,107 @@
-import React from 'react';
+/* eslint-disable react/button-has-type */
+import cn from 'classnames';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { postComment } from '../api/comments';
+import { Comment, CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  setComments: Dispatch<SetStateAction<Comment[]>>
+  selectedPost: Post | null
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  setComments,
+  selectedPost,
+}) => {
+  const defaultComment = {
+    name: '',
+    email: '',
+    body: '',
+    postId: selectedPost?.id,
+  };
+  const defaultFormError = {
+    name: false,
+    email: false,
+    body: false,
+    submittingError: false,
+  };
+
+  const [commentData, setCommentData] = useState<CommentData>(defaultComment);
+  const [isSubmiting, setSubmiting] = useState(false);
+  const [formErrors, setFormError] = useState(defaultFormError);
+  const { name, email, body } = commentData;
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement
+    | HTMLTextAreaElement>,
+  ) => {
+    setCommentData(prev => ({
+      ...prev,
+      [event.target.name]: event.target.value,
+    }));
+
+    setFormError(prev => ({
+      ...prev,
+      [event.target.name]: false,
+    }));
+  };
+
+  const addComment = async () => {
+    setSubmiting(true);
+    try {
+      const newComment = await postComment(commentData);
+
+      setComments(prev => [...prev, newComment]);
+    } catch {
+      setFormError(prev => ({
+        ...prev,
+        submittingError: true,
+      }));
+    } finally {
+      setSubmiting(false);
+    }
+  };
+
+  const handleSubmit = (event:React.FormEvent<HTMLFormElement>) => {
+    let hasError = false;
+
+    event.preventDefault();
+    if (!name) {
+      setFormError(prev => ({ ...prev, name: true }));
+      hasError = true;
+    }
+
+    if (!email) {
+      setFormError(prev => ({ ...prev, email: true }));
+      hasError = true;
+    }
```

Comment:
would be nice to evaluate email format as well
<img width="1168" alt="Monosnap Dynamic list of posts 2023-02-03 09-38-36" src="https://user-images.githubusercontent.com/39308717/216542230-23d1229a-7352-48c2-af6a-5708d92437de.png">